build.sh

```diff
@@ -1,66 +1,101 @@
 }
 
-# download module definitions for libmongoc/libbson
-[[ -d Sources/libbson ]] || git clone --depth 1 https://github.com/mongodb/swift-bson Sources/libbson
-[[ -d Sources/libmongoc ]] || git clone --depth 1 https://github.com/mongodb/swift-mongoc Sources/libmongoc
+download_and_combine() {
+    local variant=$1
+    local variant_os=${variant}os
+    local variant_os_tmp=${variant_os}-tmp
+    local variant_simulator=${variant}simulator
+    local variant_simulator_tmp=${variant_simulator}-tmp
 
-# vendor in MongoSwift
-if [ ! -d Sources/MongoSwift ]; then
-  curl -L https://api.github.com/repos/mongodb/mongo-swift-driver/tarball > mongo-swift.tgz
-  mkdir mongo-swift
-  tar -xzf mongo-swift.tgz -C mongo-swift --strip-components 1
-  cp -r mongo-swift/Sources/MongoSwift Sources/MongoSwift
-  # TODO: copy tests
+    if [ ! -d $variant_os ]; then
+        download_mobile_sdk $variant_os_tmp $(mobile_sdk_url ${variant_os})
+        fix_mongoc_symlinks $variant_os_tmp
 
-  rm -rf mongo-swift mongo-swift.tgz
-fi
+        download_mobile_sdk $variant_simulator_tmp $(mobile_sdk_url ${variant_simulator})
+        fix_mongoc_symlinks $variant_simulator_tmp
 
-# download mobile SDKs
-mkdir -p MobileSDKs && cd MobileSDKs
+        # create shared include path
+        cp -r $variant_os_tmp/include .
 
-if [ ! -d iphoneos ]; then
-  download_mobile_sdk "iphoneos-tmp" $MOBILE_SDK_IPHONEOS
-  fix_mongoc_symlinks "iphoneos-tmp"
+        # create combined SDK library paths
+        mkdir -p ${variant_os}/lib
 
-  download_mobile_sdk "iphonesimulator-tmp" $MOBILE_SDK_IPHONESIMULATOR
-  fix_mongoc_symlinks "iphonesimulator-tmp"
+        echo "merging architectures into universal dylibs..."
+        for lib in $variant_os_tmp/lib/*.dylib; do
+            base_lib=$(basename "$lib")
+            lipo $variant_os_tmp/lib/${base_lib} $variant_simulator_tmp/lib/${base_lib} -output ${variant_os}/lib/${base_lib} -create
+        done
 
-  # create shared include path
-  cp -r iphoneos-tmp/include .
+        # cleanup
+        rm -rf $variant_os_tmp
+        rm -rf $variant_simulator_tmp
+    fi
+}
 
-  # create combined SDK library paths
-  mkdir -p iphoneos/lib
+# download module definitions for libmongoc/libbson
+[[ -d Sources/libbson ]] || git clone --depth 1 https://github.com/mongodb/swift-bson Sources/libbson
+[[ -d Sources/libmongoc ]] || git clone --depth 1 https://github.com/mongodb/swift-mongoc Sources/libmongoc
+
+# download mobile SDKs
+mkdir -p MobileSDKs && cd MobileSDKs
+download_and_combine "iphone"
+download_and_combine "appletv"
+cd ..
 
-  echo "merging architectures into universal dylibs..."
-  for lib in iphoneos-tmp/lib/*.dylib; do
-    base_lib=$(basename "$lib")
-    lipo iphoneos-tmp/lib/${base_lib} iphonesimulator-tmp/lib/${base_lib} -output iphoneos/lib/${base_lib} -create
-  done
+# vendor in MongoMobile
+if [ ! -d MobileSDKs/include/mongo/embedded-v1/mongo_embedded ]; then
```

Comment:
@edaniels This is where we are vendoring in MongoMobile until a better solution is present.